runhouse/resources/serverless/aws_lambda_function.py

```diff
@@ -85,76 +91,41 @@ def __init__(
         .. note::
                 To create an AWS lambda resource, please use the factory method :func:`aws_lambda_function`.
         """
-        if isinstance(fn, Callable):
-            handler_function_name = fn.__name__
-            fn_pointers = Function._extract_pointers(fn, reqs=env or [])
-            paths_to_code = self._paths_to_code_from_fn_pointers(fn_pointers)
-            args_names = [
-                param.name for param in inspect.signature(fn).parameters.values()
-            ]
-        elif fn_pointers:
-            handler_function_name = fn_pointers[2]
-            paths_to_code = self._paths_to_code_from_fn_pointers(fn_pointers)
-        else:
-            fn_pointers = None
-            if handler_function_name is None or paths_to_code is None:
-                raise ValueError(
-                    "`handler_function_name` and `paths_to_code` must be provided if `fn` is not provided"
-                )
-
-        if name is None:
-            name = handler_function_name.replace(".", "_")
 
         super().__init__(
```

Comment:
Just noting that the `access` attribute will probably need be included here eventually